examples/visual_examples/car_sequencing/run_car_seq.py

```diff
@@ -14,44 +14,87 @@
 from PIL import Image, ImageDraw, ImageFont
 import datetime
 
-# All data related to the sequencing
-nOptions = 5
-nCarConfigs = 6
-at_most = [1, 2, 2, 2, 1]
-per_slots = [2, 3, 3, 5, 5]
-demand = [1,1,2,2,2,2]
-nCars = sum(demand)
-nSlots = nCars
-requires = [
-[1, 0, 1, 1, 0],
-[0, 0, 0, 1, 0],
-[0, 1, 0, 0, 1],
-[0, 1, 0, 1, 0],
-[1, 0, 1, 0, 0],
-[1, 1, 0, 0, 0]]
-
-# Datetime generators for printing in visualisation
-datetime_object = datetime.datetime.strptime('14:00', '%H:%M')
-date_list = [datetime_object + datetime.timedelta(minutes=15*x) for x in range(0, nSlots)]
-print_dates = [d.strftime('%H:%M') for d in date_list]
-
-# Decision variables
-line = intvar(1, nCarConfigs, shape = nSlots)
-setup = boolvar(shape = (nSlots, nOptions))
-
-model = Model(
-    [sum((x == (i+1)) for x in line) == demand[i] for i in range(nCarConfigs)] # The amount of each type of car in the timetable has to be equal to the demand for that type
-)
-
-for o in range(nOptions):
-    model += [sum(setup[s2][o] == True for s2 in range(s, s + per_slots[o])) <= at_most[o] for s in range(nSlots - per_slots[o])] # Check that no more than "at most" car properties are used per "per_slots" slots
-
-requires = cpm_array(requires) # So it can be indexed by a variable
-model += [setup[s,o] == requires[line[s]-1,o] for s in range(nSlots) for o in range(nOptions)] # Make sure that the properties in the timetable correspond to those per car type
-
-if model.solve():
-    print(line.value())
-    print(setup.value())
+def run():
+    # All data related to the sequencing
+    nOptions = 5
+    nCarConfigs = 6
+    at_most = [1, 2, 2, 2, 1]
+    per_slots = [2, 3, 3, 5, 5]
+    demand = [1,1,2,2,2,2]
+    nCars = sum(demand)
+    nSlots = nCars
+    requires = [
+    [1, 0, 1, 1, 0],
+    [0, 0, 0, 1, 0],
+    [0, 1, 0, 0, 1],
+    [0, 1, 0, 1, 0],
+    [1, 0, 1, 0, 0],
+    [1, 1, 0, 0, 0]]
+
+    (model, vars) = model_sequence(nCarConfigs, nSlots, nOptions, demand, per_slots, at_most, requires)
+
+    if model.solve():
+        for v in vars:
+            print(v + ":\n" + str(vars[v].value()))
+        visualize_sequence(vars, nSlots, nOptions, nCarConfigs, requires, demand, at_most, nCars, per_slots)
+
+def model_sequence(nCarConfigs, nSlots, nOptions, demand, per_slots, at_most, requires):
+    # Decision variables
```

Comment:
Please add more of the problem description in the comments..
what does 'line' represent (the sequence of cars?)
what does 'setup' represent...